lib/dao.js

```diff
@@ -2450,128 +2489,107 @@ DataAccessObject.prototype.updateAttributes = function updateAttributes(data, op
   Model.notifyObserversOf('before save', context, function(err, ctx) {
     if (err) return cb(err);
     data = ctx.data;
-
     if (strict && !allowExtendedOperators) {
-      var props = Model.definition.properties;
-      var keys = Object.keys(data);
-      var result = {}, key;
-      for (var i = 0; i < keys.length; i++) {
-        key = keys[i];
-        if (props[key]) {
-          result[key] = data[key];
-        } else if (strict === 'throw') {
-          cb(new Error('Unknown property: ' + key));
-          return;
-        } else if (strict === 'validate') {
-          inst.__unknownProperties.push(key);
+      strictCheck(self.constructor, strict, data, inst, strictChkCB);
+    } else {
+      strictChkCB(null, removeUndefined(data));
+    }
+
+    function strictChkCB(err, data) {
```

Comment:
Please use a name that describes what the function is actually doing, e.g. `validateAndSave`.